Editorial fixes for recovery (#4213)
quicwg/base-drafts: draft-ietf-quic-recovery.md

such cases, an endpoint SHOULD subtract such local delays from its RTT sample
until the handshake is confirmed.

Similar to {{?RFC6298}}, smoothed_rtt and rttvar are computed as follows.

> An endpoint needs to initialize the RTT estimator during connection

Reviewer: ```suggestion
An endpoint needs to initialize the RTT estimator during connection
```
```suggestion
An endpoint initializes the RTT estimator during connection
```

establishment and when the estimator is reset during connection migration

> (Section 9.4 of {{QUIC-TRANSPORT}}). Before any RTT samples are available for a
> new path or when the estimator is reset, the estimator is initialized using an
> initial RTT value of 333ms; see {{pto-handshake}}.

Reviewer: This is slightly different from what we say in handshakes and new paths(pto-handshake).  I'd suggest referencing that rather than re-specifying 333ms here.
```suggestion
(Section 9.4 of {{QUIC-TRANSPORT}}). Before any RTT samples are available for a
new path or when the estimator is reset, the estimator is initialized using the
initial RTT; see {{pto-handshake}}.
```

FYI, existing text is:
   Resumed connections over the same network MAY use the previous
   connection's final smoothed RTT value as the resumed connection's
   initial RTT.  When no previous RTT is available, the initial RTT
   SHOULD be set to 333ms, resulting in a 1 second initial timeout, as
   recommended in [RFC6298].

Author: Fair enough.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.

> An endpoint needs to bootstrap the RTT estimator during connection establishment

Reviewer: I'm not sure if initialize is right, but bootstrap is not a commonly used word.

```suggestion
An endpoint needs to initialize the RTT estimator during connection establishment
```

and when the estimator is reset during connection migration (Section 9.4 of
{{QUIC-TRANSPORT}}).

When there are no RTT samples for a new network path, and on the first
subsequent RTT sample for the new network path, smoothed_rtt and rttvar are set
as follows:

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

> Before any RTT samples are available for a new path or when the estimator is

Reviewer: How about moving these two paragraph above the paragraph starting with "When there are no RTT samples..." and reworking the text a bit as a result?

Author: I actually did exactly that, and realized that it's awkward to use `rtt_sample` in that structure, since it's not yet been used. But yeah, this still isn't great. I'll work on it a bit.

Author: Ok, done.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.

An endpoint needs to initialize the RTT estimator during connection establishment
and when the estimator is reset during connection migration (Section 9.4 of

> {{QUIC-TRANSPORT}}).

Reviewer: ```suggestion
An endpoint initializes the RTT estimator during connection establishment
and when the estimator is reset during connection migration (Section 9.4 of
{{QUIC-TRANSPORT}}).
```

Before any RTT samples are available for a new path or when the estimator is

> reset, the estimator is initialized using an initial RTT value of 333ms; see

Reviewer: ```suggestion
Before any RTT samples are available for a new path, the estimator is
initialized using an initial RTT value of 333ms; see
```

{{pto-handshake}}. On the first subsequent RTT sample for the network path, the
estimator is reset using that sample. This ensures that the estimator retains no
history of past samples.

In both of these cases, smoothed_rtt and rttvar are set as follows:

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

> In the computation above, rtt_sample is initialized to the initial RTT value,

Reviewer: I'd prefer to remove this paragraph.
